### ISSUE TITLE
Static analysis unexplainably failing

### ISSUE THREAD

**Issue by renggli** [dart-lang/pub-dev#4842]
URL: https://pub.dev/packages/petitparser/score
Screenshot: https://user-images.githubusercontent.com/131610/120111299-acd46d00-c171-11eb-94da-8ced3aa57915.png

As originally reported by https://github.com/petitparser/dart-petitparser/issues/111, the static analysis on pub.dev of my package is suddenly failing with `dartanalyzer exception: RangeError (index): Invalid value: Not in inclusive range 0..12: 13`.

I cannot reproduce the issue on the published code with the same version of Dart:

```bash
$ git checkout 4.1.0
Note: switching to '4.1.0'.

$ dart --version
Dart SDK version: 2.13.1 (stable) (Fri May 21 12:45:36 2021 +0200) on "macos_x64"

$ dartanalyzer .
Analyzing petitparser...
No issues found!
```

**Comment by isoos** [comment]
https://github.com/dart-lang/pana/issues/893

**Comment by isoos** [comment]
Did you do `pub upgrade` or a clean git clone + `pub get`? With that, I got the same exception from the `dartanalyzer` tool.

**Comment by renggli** [comment]
Thanks, forgot to run `pub upgrade`. Like this I get the exception locally too ...

```
Analyzing petitparser...
Unhandled exception:
RangeError (index): Invalid value: Not in inclusive range 0..12: 13
#0      List.[] (dart:core-patch/array.dart:268:52)
#1      BestPracticesVerifier._targetKindsFor (package:analyzer/src/error/best_practices_verifier.dart:1601:38)
#2      BestPracticesVerifier.visitAnnotation (package:analyzer/src/error/best_practices_verifier.dart:253:17)
#3      AnnotationImpl.accept (package:analyzer/src/dart/ast/ast.dart:302:50)
#4      NodeListImpl.accept (package:analyzer/src/dart/ast/ast.dart:7517:20)
#5      AnnotatedNodeImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:145:17)
#6      ClassDeclarationImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:1537:11)
#7      RecursiveAstVisitor.visitClassDeclaration (package:analyzer/dart/ast/visitor.dart:711:10)
#8      BestPracticesVerifier.visitClassDeclaration (package:analyzer/src/error/best_practices_verifier.dart:323:13)
#9      ClassDeclarationImpl.accept (package:analyzer/src/dart/ast/ast.dart:1514:50)
#10     NodeListImpl.accept (package:analyzer/src/dart/ast/ast.dart:7517:20)
#11     CompilationUnitImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:2112:21)
#12     RecursiveAstVisitor.visitCompilationUnit (package:analyzer/dart/ast/visitor.dart:735:10)
#13     CompilationUnitImpl.accept (package:analyzer/src/dart/ast/ast.dart:2105:50)
#14     LibraryAnalyzer._computeHints (package:analyzer/src/dart/analysis/library_analyzer.dart:276:10)
#15     LibraryAnalyzer.analyzeSync.<anonymous closure> (package:analyzer/src/dart/analysis/library_analyzer.dart:158:9)
#16     _LinkedHashMapMixin.forEach (dart:collection-patch/compact_hash.dart:397:8)
#17     LibraryAnalyzer.analyzeSync (package:analyzer/src/dart/analysis/library_analyzer.dart:157:13)
#18     LibraryAnalyzer.analyze (package:analyzer/src/dart/analysis/library_analyzer.dart:104:12)
#19     AnalysisDriver._computeAnalysisResult.<anonymous closure> (package:analyzer/src/dart/analysis/driver.dart:1315:63)
#20     PerformanceLog.run (package:analyzer/src/dart/analysis/performance_logger.dart:32:15)
#21     AnalysisDriver._computeAnalysisResult (package:analyzer/src/dart/analysis/driver.dart:1286:20)
#22     AnalysisDriver._computeErrors (package:analyzer/src/dart/analysis/driver.dart:1369:26)
#23     AnalysisDriver.performWork (package:analyzer/src/dart/analysis/driver.dart:987:20)
#24     AnalysisDriverScheduler._run (package:analyzer/src/dart/analysis/driver.dart:2006:24)
<asynchronous suspension>

#0      AnalysisDriver._computeAnalysisResult.<anonymous closure> (package:analyzer/src/dart/analysis/driver.dart:1344:9)
#1      PerformanceLog.run (package:analyzer/src/dart/analysis/performance_logger.dart:32:15)
#2      AnalysisDriver._computeAnalysisResult (package:analyzer/src/dart/analysis/driver.dart:1286:20)
#3      AnalysisDriver._computeErrors (package:analyzer/src/dart/analysis/driver.dart:1369:26)
#4      AnalysisDriver.performWork (package:analyzer/src/dart/analysis/driver.dart:987:20)
#5      AnalysisDriverScheduler._run (package:analyzer/src/dart/analysis/driver.dart:2006:24)
<asynchronous suspension>
```

I guess this can be closed as a duplicate of dart-lang/pana#893?
